rfd/0004-device-attestation.md

```diff
@@ -0,0 +1,66 @@
+---
+authors:  Ben Arent (ben@gravitational.com), Alexander Klizhentas (sasha@gravitational.com),
+state: draft
+---
+
+# RFD 4 - Device Attestation
+
+## What?
+
+Device attestation acts as a form of contextual access where the device must pass
+an inspection before the user is granted access, even if they have the correct
+Teleport Login.
+
+Teleport uses SSO to verify customer identity and set RBAC rules to set access,
+we’ve companies that not only want to verify if it’s the user but also that it’s
+coming from an approved endpoint.
+
+## Why?
+
+Teleport customers want to only allow `tsh` to work on corporate supplied laptops
+that are patched and have the up-to-date corporate policies on them, through a
+EPP, MDM, SCEP or internal service that’ll only allow issued devices ( laptops
+/ workstations ) to access Teleport.
+
+## Device Setup
+
+The IT Department will update a device via MDM management, and shall provide a
+certificate to end user devices.
+
+Trusted device issued a long lived X509 certificate by a separate certificate authority
+used for all devices. The certificate metadata includes the device ID, owner and
+any other information.
+
+For this RFD, we'll assume they'll be an out of band host scanning and certificate
+distribution.
+
+[Certificate Scanning](diagrams/0004/certificates-out-of-band.png)
+
+## Login Experience
+
+On Login tsh will contact the auth server using the long lived device certificate,
```

Comment:
The biggest question for this design is which device cert stores we need to support.
AFAIK every OS has its own way of storing such secrets and there isn't a consistent API.